### ISSUE TITLE
Error summary doesn't receive keyboard focus when rendered

### ISSUE THREAD

**Issue by damian** [nhsuk/nhsuk-frontend#702]
When a form has validation errors, and the error summary is shown I expected it to receive keyboard focus(otherwise why has it got a `tabindex="-1"` but it's not. I believe this to be the reason why the GDS error summary has focus attached [via JavaScript](https://github.com/alphagov/govuk-frontend/blob/803da2c92d1a73401ac506b2dc8a393fae940508/src/govuk/components/error-summary/error-summary.js#L14).

I'm happy to contribute a PR to address this if it needs a similar fix to the GDS one 😄

**Comment by AdamChrimes** [comment]
Thanks for raising this issue @damian, we didn't include the Error summary JavaScript as we usually left apps on the NHS website to handle the error logic individually, but now the library has grown it makes sense to add it to the library.

Happy for you to open a PR to address this, below is the code we use on some of the applications on the NHS website.

```
// Add autofocus and error focusing for error summary
document.addEventListener('DOMContentLoaded', function () {
  // Error summary component
  var summary = document.querySelector('.nhsuk-error-summary');
  if (summary) {
    // Focus error summary component if it exists
    summary.focus();
    // Error summary links
    var summaryLinks = document.querySelectorAll('.nhsuk-error-summary__list a');
    // Attach click event to each error summary link
    for (var i = 0; i < summaryLinks.length; i++) {
      summaryLinks[i].addEventListener('click', function (event) {
        event.preventDefault();
        // Input component to focus
        var input = document.querySelector(event.target.hash);
        // Focus input component if it exists
        if (input) input.focus();
      });
    };
  };
});
```

cc: @tomdoughty

**Comment by tomdoughty** [comment]
Yeah would be good to get that in ES6 and into the library... it has been copy and pasted too many times!
